src/stories/ProfileHeader.stories.tsx

```diff
@@ -78,17 +83,82 @@ export const Example: Story = {
     const [profileHeaderPopupOpen, setProfileHeaderPopupOpen] = useState(
       args.profileHeaderPopupOpen
     );
+    const [tabKeyPressedCount, setTabKeyPressedCount] = useState(0);
+
+    const ref = useRef<HTMLDivElement>(null);
+
+    const handleClickOutsidePopup = (ev: any) => {
+      ev.preventDefault();
+
+      if (ref.current && !ref.current?.contains(ev.target as Node)) {
+        setProfileHeaderPopupOpen(() => {
+          args.profileHeaderPopupOpen = !profileHeaderPopupOpen;
+          return !profileHeaderPopupOpen;
+        });
+      }
+    };
+
+    const handleKeyDown = (ev: any) => {
+      if (ev.which == 9) {
+        const popupMenuItems: any = document.getElementsByClassName(
+          "profile-header-popup-menu-item"
+        );
+
+        if (
+          popupMenuItems[tabKeyPressedCount] === document.activeElement &&
+          tabKeyPressedCount === 3
+        ) {
+          ev.preventDefault();
+
+          setProfileHeaderPopupOpen(() => {
+            args.profileHeaderPopupOpen = !profileHeaderPopupOpen;
+            return !profileHeaderPopupOpen;
+          });
+
+          setTabKeyPressedCount(0);
+        } else if (
+          popupMenuItems[tabKeyPressedCount] === document.activeElement
+        ) {
+          setTabKeyPressedCount(tabKeyPressedCount + 1);
+        }
+      }
+    };
+
+    useEffect(() => {
+      document.addEventListener("click", handleClickOutsidePopup, true);
+
+      return () => {
+        document.removeEventListener("click", handleClickOutsidePopup, true);
+      };
+    });
+
+    useEffect(() => {
+      document.addEventListener("keydown", handleKeyDown, true);
+
+      return () => {
+        document.removeEventListener("keydown", handleKeyDown, true);
+      };
+    });
 
     if (args.menuItems) {
-      args.menuItems[menuItems.length - 1].onClick = () =>
+      args.menuItems[menuItems.length - 1].onClick = () => {
         setProfileHeaderPopupOpen(() => {
```

Comment:
Something is funky with this onclick handler.  It doesn't close the popup menu on the second click.  Ultimately, I think this click handling should be a default behavior built into the component.